library/alloc/src/vec/in_place_collect.rs

```diff
@@ -372,8 +373,10 @@ where
         // - unlike most internal iteration methods, it only takes a &mut self
         // - it lets us thread the write pointer through its innards and get it back in the end
         let sink = InPlaceDrop { inner: dst_buf, dst: dst_buf };
-        let sink =
-            self.try_fold::<_, _, Result<_, !>>(sink, write_in_place_with_drop(end)).unwrap();
+        let sink = match self.try_fold::<_, _, Result<_, !>>(sink, write_in_place_with_drop(end)) {
+            Ok(sink) => sink,
+            Err(never) => match never {},
+        };
```

Comment:
Why not change the `unwrap` to `into_ok`?

Reply:
I didn't know about `Result::into_ok` existed. I can update the PR to use that instead if you'd like.

Reply:
yes that would be more readable